File: frontend/src/services/userService.js
import axios from 'axios';

const API_URL = process.env.REACT_APP_API_URL || 'http://localhost:8080/api';

export const userService = {
    getAllUsers: async () => {
        const response = await axios.get(`${API_URL}/users`);
        return response.data;
    },

    getUserById: async (id) => {
        const response = await axios.get(`${API_URL}/users/${id}`);
        return response.data;
    },

    createUser: async (user) => {
        const response = await axios.post(`${API_URL}/users`, user);
        return response.data;
    },

    updateUser: async (id, user) => {
        const response = await axios.put(`${API_URL}/users/${id}`, user);
        return response.data;
    },

    deleteUser: async (id) => {
        await axios.delete(`${API_URL}/users/${id}`);
    }
};

File: backend/src/main/java/com/example/service/UserService.java
package com.example.service;

import com.example.exception.ResourceNotFoundException;
import com.example.model.User;
import com.example.repository.UserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class UserService {
    
    @Autowired
    private UserRepository userRepository;
    
    @Autowired
    private PasswordEncoder passwordEncoder;
    
    public List<User> getAllUsers() {
        return userRepository.findAll();
    }
    
    public Optional<User> getUserById(Long id) {
        return userRepository.findById(id);
    }
    
    public User createUser(User user) {
        if (userRepository.existsByEmail(user.getEmail())) {
            throw new RuntimeException("Email already exists");
        }
        // Define senha padrão "123" se não foi fornecida
        if (user.getPassword() == null || user.getPassword().isEmpty()) {
            user.setPassword(passwordEncoder.encode("123"));
        }
        return userRepository.save(user);
    }
    
    public User updateUser(Long id, User userDetails) {
        User user = userRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("User not found"));
        
        user.setName(userDetails.getName());
        user.setEmail(userDetails.getEmail());
        
        return userRepository.save(user);
    }
    
    public void deleteUser(Long id) {
        User user = userRepository.findById(id)
                .orElseThrow(() -> new ResourceNotFoundException("User", "id", id));
        userRepository.delete(user);
    }
}
